enterprise/api/v1/serializers.py

```diff
@@ -214,3 +215,106 @@ def update(self, instance, validated_data):
     user = UserSerializer(read_only=True)
     enterprise_customer = EnterpriseCustomerSerializer(read_only=True)
     data_sharing_consent = UserDataSharingConsentAuditSerializer(many=True, read_only=True)
+
+
+class EnterpriseCourseCatalogReadOnlySerializer(serializers.Serializer):
+    """
+    Serializer for enterprise customer catalog.
+    """
+    # pylint: disable=invalid-name
+    id = serializers.IntegerField(read_only=True, help_text=_('Enterprise course catalog primary key.'))
+    name = serializers.CharField(help_text=_('Catalog name'))
+    query = serializers.CharField(help_text=_('Query to retrieve catalog contents'))
+    courses_count = serializers.IntegerField(read_only=True, help_text=_('Number of courses contained in this catalog'))
+    viewers = serializers.ListField(
+        allow_null=True, allow_empty=True, required=False,
+        help_text=_('Usernames of users with explicit access to view this catalog'),
+        style={'base_template': 'input.html'},
+        child=serializers.CharField(),
+    )
+
+    def create(self, validated_data):
+        """
+        Do not perform any operations for state changing requests.
+        """
+        pass
+
+    def update(self, instance, validated_data):
+        """
+        Do not perform any operations for state changing requests.
+        """
+        pass
+
+
+class EnterpriseCatalogCoursesReadOnlySerializer(serializers.Serializer):
+    """
+    Serializer for enterprise customer catalog courses.
+    """
+    count = serializers.IntegerField(read_only=True, help_text=_('Total count of catalog courses.'))
+    next = serializers.CharField(read_only=True, help_text=_("URL to fetch next page of courses."))
+    previous = serializers.CharField(read_only=True, help_text=_("URL to fetch previous page of courses."))
+    results = serializers.ListField(read_only=True, help_text=_("list of courses."))
+
+    def update_enterprise_courses(self, request):
+        """
+        This method adds enterprise specific metadata for each course.
+
+        We are adding following field in all the courses.
+            tpa_hint: a string for identifying Identity Provider.
+        """
+        courses = []
+        tpa_hint = utils.get_identity_provider_id_for_user(request.user)
+
+        global_context = {
```

Comment:
@haikuginger `global_context` contains key-value pairs that will be added to all the courses and the `tpa_hint` argument is used by `update_course` method to construct track selection url. 
Currently, `global_context` happens to have `tpa_hint` as key-value pair but it is likely that we change it to something else or remove it in the future. And I do not think it would be a good idea to put any restrictions on `global_context` to have certain value. That is why I passed `tpa_hint` as separate argument instead of retrieving it from `global_context`.